plugins/hydra_ray_launcher/hydra_plugins/hydra_ray_launcher/ray_aws_config.py

```diff
@@ -0,0 +1,101 @@
+# Copyright (c) Facebook, Inc. and its affiliates. All Rights Reserved
+from dataclasses import dataclass, field
+from typing import Any, Dict, List
+
+from hydra.core.config_store import ConfigStore
+from hydra.types import ObjectConf
+
+
+def default_field(obj):
+    return field(default_factory=lambda: obj)
+
+
+# This class maps to a Ray cluster config. To get more info on each field, please check:
+# https://github.com/ray-project/ray/blob/master/python/ray/autoscaler/aws/example-full.yaml
+# Note: You cannot add arbitrary field not supported by Ray in this conf.
+@dataclass
+class RayClusterConfig:
+    cluster_name: str = "default"
+    min_workers: int = 0
+    max_workers: int = 1
+    initial_workers: int = 0
+    autoscaling_mode: str = "default"
+    docker: Dict[str, Any] = default_field(
+        {"image": "", "container_name": "", "pull_before_run": True, "run_options": []}
+    )
```

Comment:
You can also create a @dataclass for the Docker config and use it instead of the free form dictionary.

If you still need free form key/values, you can have your dataclass extend Dict[str, Any], this way you get both type safety for known fields and the ability to add unknown fields.